secedgar/cli.py

```diff
@@ -48,7 +48,9 @@ def date_cleanup(date):
               help="""Directory where files will be saved.
               Defaults to directory from which CLI is being executed.""",
               default=os.getcwd(), type=str)
-def filing(lookups, ftype, start, end, count, directory):
+@click.option('-u', '--user-agent',
```

Comment:
I've changed the approach where the `--user-agent` flag is added to each command, `daily` and `filings`.

At the cost of duplicating the option for each command, it is simpler compared to the `context` object approach I tried yesterday: https://github.com/sec-edgar/sec-edgar/pull/200#discussion_r612000577. One key reason why it is simpler is because if we use the `context` object approach, we'll have to somehow construct and pass the `context` object here: https://github.com/sec-edgar/sec-edgar/blob/master/secedgar/tests/test_cli.py#L10

What do you guys think of this approach? If you guys prefer the `context` object approach I can try that as well.

Reply:
Why wouldn't that be included in the user_input variable? Has been a while since I have worked on the CLI, so I will need to see how this works.

Reply:
That can indeed be made to work, however `test_cli.py` will have to be updated a bit - it needs to test `cli` rather than `filings` and `daily`. The reason being that in my old approach I added the `--user-agent` option at the `cli` rather than `filings` and `daily`.

It's going to look like this more or less:

```
import pytest
from click.testing import CliRunner
from secedgar.cli import cli
from secedgar.exceptions import FilingTypeError


def run_cli_command(cli, user_input, directory, catch_exceptions=False):
    ...


def check_bad_inputs(cli, user_input, expected_exception, directory):
    ...


class TestCLI:
    @pytest.mark.parametrize(
        "user_input,expected_exception",
        [
            ("-u 'User Agent (email@example.com)' filing -l aapl msft Facebook", SystemExit),  # missing filing type
            ("-u 'User Agent (email@example.com)' filing -l aapl -t null", FilingTypeError),  # unrecognized filing type
            ("-u 'User Agent (email@example.com)' filing -l aapl -t FILING_10Q -n abc", SystemExit),  # count is not int
            ("-u 'User Agent (email@example.com)' filing -l aapl -t FILING_10Q -n 0", ValueError)  # no filings available if 0 picked
        ]
    )
    def test_filing_bad_inputs(self, user_input, expected_exception, tmp_data_directory):
        check_bad_inputs(cli, user_input, expected_exception, tmp_data_directory)

@pytest.mark.parametrize(
        "user_input,expected_exception",
        [
            ("-u 'User Agent (email@example.com)' daily", SystemExit),
            ("-u 'User Agent (email@example.com)' daily -d 2020", ValueError)
        ]
    )
    def test_daily_bad_inputs(self, user_input, expected_exception, tmp_data_directory):    
        check_bad_inputs(cli, user_input, expected_exception, tmp_data_directory)
```

What do you think?

Do you prefer the approach we already have (where `--user-agent` is added twice to `filings` and `daily`), or approach I proposed here (where we use context object which allows us to add `--user-agent` only once (to `cli`)?

Personally I like the proposed approach since you don't duplicate the `--user-agent` switch. Plus testing at the `cli` level is more end-to-end, ie., you test from the root rather than from the inner commands. However I'm gonna leave the decision up to you as the maintainer.

Reply:
I think the proposed approach makes sense. Looks like you have already done the hard work with testing as well, which is awesome!

Reply:
Cool! Let's see what we can come up with.